main.cpp

```diff
@@ -115,17 +133,17 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T t1( 5, "a");                                             //6
+    T t2( 7, "b");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
+    C f;                                            //7
+    auto* smaller = f.compare(&t1, &t2);                              //8
     std::cout << "the smaller one is << " << smaller->name << std::endl; //9
```

Comment:
What do we always do before using pointers?
- [ ] revise

Reply:
I have no idea